File: Backend_Springboot/cab-booking/src/main/java/com/wheelshare/cabbooking/entity/Driver.java
package com.wheelshare.cabbooking.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "drivers")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Driver {

    @Id
    @Column(name = "driver_id")
    private Integer driverId;

    @OneToOne
    @MapsId
    @JoinColumn(name = "driver_id")
    private User user;

    @Column(nullable = false, length = 50)
    private String licenseNumber;

    @Column(name = "is_verified")
    private Boolean isVerified = false;

    @Column(name = "is_available")
    private Boolean isAvailable = false;

    @Column(name = "current_latitude")
    private Double currentLatitude;

    @Column(name = "current_longitude")
    private Double currentLongitude;
}

File: Backend_DotNet/WheelShare_dotNet/WheelShare_dotNet/Models/Driver.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WheelShare_dotNet.Models
{
    [Table("Drivers")]
    public class Driver
    {

        [Key, ForeignKey("User")]
        public int DriverId { get; set; }

        [Required, MaxLength(50)]
        public string LicenseNumber { get; set; }

        public bool IsVerified { get; set; } = false;
        public bool IsAvailable { get; set; } = false;

        public double CurrentLatitude { get; set; }
        public double CurrentLongitude { get; set; }

        public User User { get; set; }

        public ICollection<Vehicle> Vehicles { get; set; }
    }
}

